### PR TITLE
Backport Flex-specific error messages in controller shortcuts to 3.4

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/25133#issuecomment-354551079 
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by weaverryan — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
👍 , but why not 3.4?

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php [hunk 1]
```diff
@@ -153,7 +153,7 @@ protected function file($file, string $fileName = null, string $disposition = Re
     protected function addFlash(string $type, string $message)
     {
         if (!$this->container->has('session')) {
-            throw new \LogicException('You can not use the addFlash method if sessions are disabled.');
+            throw new \LogicException('You can not use the addFlash method if sessions are disabled. Enable them in config/packages/framework.yaml.');
```
Comment (dunglas) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php]: This is not consistent with other message (no quotes but dot at the end).

**Comment by chalasr** [comment]
> 👍 , but why not 3.4?

no objection for me, 4.0 was the target of https://github.com/symfony/symfony/pull/25133. 
The trait could also give these hints only if `class_exists(Flex::class)`, not sure it's worth it though, let me know wdyt.

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php (hunk 1 above)
Comment (chalasr) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php]: fixed

**Comment by nicolas-grekas** [comment]
As a general rule of thumb, almost all PRs that target 4.0 should really target 3.4: bothem versions share the same bugs (with rare exception.)

**Review by xabbuh — APPROVED** [review]

**Comment by chalasr** [comment]
rebased

**Comment by chalasr** [comment]
Note that symfony-standard exists in 3.4, these suggestions would be wrong there.

File: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php [hunk 2]
```diff
@@ -194,7 +194,7 @@ protected function file($file, $fileName = null, $disposition = ResponseHeaderBa
     protected function addFlash($type, $message)
     {
         if (!$this->container->has('session')) {
-            throw new \LogicException('You can not use the addFlash method if sessions are disabled.');
+            throw new \LogicException('You can not use the addFlash method if sessions are disabled. Enable them in "config/packages/framework.yaml"');
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php]: Error messages like this -> *Enable them in "config/packages/framework.yaml"* could be problematic for 3.4 app ... which probably use the same dir layout that 3.3. The new layout is only generally used in 4.0 and newer.

Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php]: Most of the exceptions will probably never be shown if you depend on the Standard Edition (this one is indeed a bit different, but we could additionally check for the `FullStack` class).

**Review by nicolas-grekas — APPROVED** [review]
(but missing dot at the end of each message)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**